### ISSUE TITLE
After expand, mode change to normal. (from select pooups)

### ISSUE THREAD

**Issue by iranoan** [hrsh7th/vim-vsnip#255]
## absturuct
After expand, mode change to normal.

### \~/.vim/vimrc
```
call plug#begin()
Plug 'junegunn/vim-plug'
Plug 'prabirshrestha/asyncomplete.vim'
Plug 'prabirshrestha/asyncomplete-lsp.vim'
Plug 'prabirshrestha/vim-lsp'
Plug 'hrsh7th/vim-vsnip'
Plug 'hrsh7th/vim-vsnip-integ'
Plug 'rafamadriz/friendly-snippets'
call plug#end()
imap <expr> <C-j>   vsnip#expandable()  ? '<Plug>(vsnip-expand)'         : '<C-j>'
smap <expr> <C-j>   vsnip#expandable()  ? '<Plug>(vsnip-expand)'         : '<C-j>'
imap <expr> <C-l>   vsnip#available(1)  ? '<Plug>(vsnip-expand-or-jump)' : '<C-l>'
smap <expr> <C-l>   vsnip#available(1)  ? '<Plug>(vsnip-expand-or-jump)' : '<C-l>'
set viminfo+=n~/.vim/viminfo
set background=dark
colorscheme desert
filetype plugin on
set laststatus=2
" The below setting has nothing to do with the issue.
" To make it easier to understand the mode in the video.
augroup InsertStatus
  autocmd!
  autocmd InsertEnter * highlight StatusLine ctermfg=red   guibg=red
  autocmd InsertLeave * highlight StatusLine ctermfg=white guibg=white
augroup END
```
### Vim version

> LANGUAGE=C vim --version
> VIM - Vi IMproved 8.2 (2019 Dec 12, compiled Sep 13 2022 09:35:02)
> Included patches: 1-3995, 4563, 4646, 4774, 4895, 4899, 4901, 4919
> Modified by team+vim@tracker.debian.org
> Compiled by team+vim@tracker.debian.org
> Huge version with GTK3 GUI.  Features included (+) or not (-):

### OS
> $ lsb_release -a
> No LSB modules are available.
> Distributor ID:	Ubuntu
> Description:	Ubuntu 22.04.1 LTS
> Release:	22.04
> Codename:	jammy
> $ uname -a
> Linux xxxx 5.15.0-48-generic #54-Ubuntu SMP Fri Aug 26 13:26:29 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
## reproduction steps

![vsni](https://user-images.githubusercontent.com/24851657/194464934-f9c5d6e9-90ff-45b9-ad1b-9a82d0257203.gif)

1.   insert full-spell (prefix)
2.   \<C-L> (vsnip-expand-or-jump)
-> insert mode
4.   type insert
-> All will be entered.

This behavior is what I expect.

Otherwise
1.   inset prefix
2.   select from popups
3.   \<C-L> (vsnip-expand-or-jump)
-> normal mode
5.   type insert
-> No i is entered.

### add information
* The GIF-animation is Gnome Terminal, but Gvim, tilda and tmux are same.
* \<C-L> (vsnip-expand-or-jump) and \<C-J> (vsnip-expand) are same.
* \<Esc> in normal mode immediately after the latter \<C-L> (vsnip-expand-or-jump) will put you in insert mode.
It is very unnatural for Vim.

## Fix example?
vim-vsnip/plugin/vsnip.vim L.96
```
  " This is needed to keep normal-mode during 0ms to prevent CompleteDone handling by LSP Client.
  let l:maybe_complete_done = !empty(v:completed_item) && !empty(v:completed_item.user_data)
  if l:maybe_complete_done
    call timer_start(0, { -> l:ctx.callback() })
  else
```
change to
```
  " This is needed to keep normal-mode during 0ms to prevent CompleteDone handling by LSP Client.
  let l:maybe_complete_done = !empty(v:completed_item) && !empty(v:completed_item.user_data)
  if l:maybe_complete_done
    call timer_start(0, { -> l:ctx.callback() })
    if mode(1) =~# 'n'
      startinsert
    endif
  else
```

**Comment by hrsh7th** [comment]
Hm... I can't reproduce with your minimal config.


https://user-images.githubusercontent.com/629908/195973864-682be1b5-5b3c-4cc1-b98d-297b7d9008e3.mov

**Comment by iranoan** [comment]
Sorry.
mapping Not enough (<C-y>).

```
scriptversion 4
call plug#begin()
Plug 'junegunn/vim-plug'
Plug 'prabirshrestha/asyncomplete.vim'
Plug 'prabirshrestha/asyncomplete-lsp.vim'
Plug 'prabirshrestha/vim-lsp'
Plug 'hrsh7th/vim-vsnip'
Plug 'hrsh7th/vim-vsnip-integ'
Plug 'rafamadriz/friendly-snippets'
call plug#end()
imap <expr> <C-j>   vsnip#expandable()  ? '<Plug>(vsnip-expand)'         : '<C-j>'
smap <expr> <C-j>   vsnip#expandable()  ? '<Plug>(vsnip-expand)'         : '<C-j>'
imap <expr> <C-l>   vsnip#available(1)  ? '<Plug>(vsnip-expand-or-jump)' : '<C-l>'
smap <expr> <C-l>   vsnip#available(1)  ? '<Plug>(vsnip-expand-or-jump)' : '<C-l>'
" map additional beginning
imap <expr> <C-y>   vsnip#available(1)  ? '<Plug>(vsnip-expand-or-jump)' : '<C-y>'
smap <expr> <C-y>   vsnip#available(1)  ? '<Plug>(vsnip-expand-or-jump)' : '<C-y>'
" map additional ending
set viminfo+=n~/.vim/viminfo
set background=dark
" The below setting has nothing to do with the issue.
" To make it easier to understand the mode in the video.
colorscheme desert
filetype plugin on
set laststatus=2
augroup InsertStatus
  autocmd!
  autocmd InsertEnter * highlight StatusLine ctermfg=red   guibg=red
  autocmd InsertLeave * highlight StatusLine ctermfg=white guibg=white
augroup END
```

**Comment by hrsh7th** [comment]
Fixed by abfe8411a0d1c0dfea56b7221e49cbb6cf7cf4fe

**Comment by iranoan** [comment]
Thanks.
confirm to fix
